site/docs/plugins/runner.md

```diff
@@ -78,10 +80,299 @@ run(bot);
+                                              __// \\__
+                                           __/  /   \  \__
+2. send updates to workers              __/    /     \    \__
+                                     __/      /       \      \__
+                                    /        /         \        \
+3. process updates          BotWorker   BotWorker   BotWorker   BotWorker
+```
+
+grammY runner provides you with middleware that can send updates to bot workers.
+The bot workers can then receive this update and handle it.
+This way, the central bot only has to concern itself with pulling in and distributing updates among the bot workers it orchestrates.
+The actual update handling (filtering messages, sending replies, etc.) is performed by the bot workers.
+
+Let's now see how this can be used.
+
+### Using Bot Workers
+
+> Examples of this can be found in the [grammY runner repository](https://github.com/grammyjs/runner/tree/main/examples).
+
+We will start out by creating the central bot instance that fetches updates and distributes them among workers.
+Let's start by creating a file called `bot.ts` with the following content.
+
+<CodeGroup>
+  <CodeGroupItem title="TypeScript" active>
+
+```ts
+// bot.ts
+import { Bot } from "grammy";
+import { distribute, run } from "@grammyjs/runner";
+
+// Create the bot.
+const bot = new Bot("");
+
+// Optionally, sequentialize updates here.
+// bot.use(sequentialize(...))
+
+// Distribute the updates among bot workers.
+bot.use(distribute(__dirname + "/worker"));
+
+// Run the bot concurrently with multi-threading.
+run(bot);
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="JavaScript">
+
+```ts
+// bot.ts
+const { Bot } = require("grammy");
+const { distribute, run } = require("@grammyjs/runner");
+
+// Create the bot.
+const bot = new Bot("");
+
+// Optionally, sequentialize updates here.
+// bot.use(sequentialize(...))
+
+// Distribute the updates among bot workers.
+bot.use(distribute(__dirname + "/worker"));
+
+// Run the concurrently with multi-threading.
+run(bot);
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="Deno">
+
+```ts
+// bot.ts
+import { Bot } from "https://deno.land/x/grammy/mod.ts";
+import { distribute, run } from "https://deno.land/x/grammy_runner_/mod.ts";
+
+// Create the bot.
+const bot = new Bot("");
+
+// Optionally, sequentialize updates here.
+// bot.use(sequentialize(...))
+
+// Distribute the updates among bot workers.
+bot.use(distribute(new URL("./worker.ts", import.meta.url)));
+
+// Run the concurrently with multi-threading.
+run(bot);
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+Next to `bot.ts`, we create a second file called `worker.ts` (as specified on line 12 in the code above).
+This will contain the actual bot logic.
+
+<CodeGroup>
+  <CodeGroupItem title="TypeScript" active>
+
+```ts
+// worker.ts
+import { BotWorker } from "@grammyjs/runner";
+
+// Create a new bot worker.
+const bot = new BotWorker(""); // <-- pass your bot token here again
```

Comment:
I'm not so sure about this one. We want to emphasize that the token must be passed twice. Perhaps it's good to be very different here?

Reply:
Why not just `bot.token`?

Reply:
It's in a different file and you really, really don't want to add the dependency because then you'd have two files that import each other.

If anything, we could consider injecting it with the first update, but this would be less explicit, harder to understand, and more complex to implement.

Reply:
Oh then I was fucking wrong. We just don't need the `again`.

Reply:
We simply won’t.

Reply:
Can we add a note below the code snippet which points this out?

Reply:
Nvm, if anything, I'll do this in subsequent PQs. This is too late here

Reply:
Agree with @KnorpelSenf. I prefer to stick with "pass your bot token here again" to make it clear the token needs to be inserted again. Since it's already explained in the previous code, it's unlikely to confuse the reader.